apps/evm/lib/evm/builtin.ex

```diff
@@ -138,14 +134,111 @@ defmodule EVM.Builtin do
           {EVM.Gas.t(), EVM.SubState.t(), EVM.ExecEnv.t(), EVM.VM.output()}
   def run_id(gas, exec_env) do
     data = exec_env.data
-    used_gas = @g_identity_base + @g_identity_byte * MathHelper.bits_to_words(byte_size(data))
+    required_gas = @g_identity_base + @g_identity_byte * MathHelper.bits_to_words(byte_size(data))
 
-    if used_gas <= gas do
-      remaining_gas = gas - used_gas
+    if required_gas <= gas do
+      remaining_gas = gas - required_gas
 
       {remaining_gas, %EVM.SubState{}, exec_env, data}
     else
       {0, %EVM.SubState{}, exec_env, :failed}
     end
   end
+
+  @spec exp_mod(EVM.Gas.t(), EVM.ExecEnv.t()) ::
+          {EVM.Gas.t(), EVM.SubState.t(), EVM.ExecEnv.t(), EVM.VM.output()}
+  def exp_mod(gas, exec_env) do
+    length_data = Memory.read_zeroed_memory(exec_env.data, 0, 96)
```

Comment:
Why do we pass `0` and `96` here? I was looking at EIP 198 and couldn't find those numbers in particular. Do you think we could extract them into variables to give them names?

Reply:
we pass 96 because the first three numbers (lengths of modulus, base and exponent) have a size of 32 bytes. Yes, I'll extract it into a variable.